source/slang/slang-ir.h

```diff
@@ -1265,15 +1265,30 @@ struct IRRTTIType : IRType
     IR_LEAF_ISA(RTTIType);
 };
 
-struct IRAnyValueType : IRType
+struct IRAnyValueTypeBase : IRType
 {
-    IR_LEAF_ISA(AnyValueType);
+    IR_PARENT_ISA(AnyValueTypeBase)
     IRInst* getSize()
     {
         return getOperand(0);
     }
 };
 
+struct IRAnyValueType : IRAnyValueTypeBase
+{
+    IR_LEAF_ISA(AnyValueType);
+    
+};
+
+struct IRAnyValueTypeWithRTTI : IRAnyValueTypeBase
+{
+    IR_LEAF_ISA(AnyValueTypeWithRTTI);
+    IRInst* getRTTI()
+    {
+        return getOperand(1);
+    }
+};
```

Comment:
The main question I have on this PR is: why have a pass that transforms `AnyValueType` to `AnyValueTypeWithRTTI` instead of just transforming the `makeExistential` operations themselves?

The current encoding of `makeExistential(v, w)` relies on the fact that the type that needs to go into the existential can be read off of the `v` operand, and thus we don't bother making it an explicit operand.

An alternative encoding would be to have `makeExistential(v, T, w)` take the type as an explicit operand, and then a pass could replace the type operand `T` with an RTTI operand instead.

Alternatively, we could also have `makeExistentialWithRTTI(v, RTTI, w)` and have the pass transform `makeExistential` to `makeExistentialWithRTTI`.

I'm just a bit confused about why `AnyValue` needs to sacrifice simplicity and clarity for this feature, when it seems like the design issue might just be on `makeExistential`.

Reply:
That's true. Let me redo this PR. Thanks for pointing it out!